site/app/controllers/PollController.php

```diff
@@ -152,11 +154,20 @@ public function addNewPoll() {
                 $answers[] = $_POST["option_id_" . $i];
             }
         }
+
         if (count($answers) == 0) {
             $this->core->addErrorMessage("Polls must have at least one correct response");
             new RedirectResponse($this->core->buildCourseUrl(['polls']));
         }
-        $this->core->getQueries()->addNewPoll($_POST["name"], $_POST["question"], $responses, $answers, $_POST["release_date"], $orders);
+
+        $poll_id = $this->core->getQueries()->addNewPoll($_POST["name"], $_POST["question"], $responses, $answers, $_POST["release_date"], $orders);
+        $file_path = null;
+        if (isset($_FILES['image_file']) && $_FILES["image_file"]["name"] !== "") {
+            $file = $_FILES["image_file"];
+            $file_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "uploads", "polls", "poll_image_" . $poll_id . "_" . $_FILES["image_file"]["name"]);
+            copy($file["tmp_name"], $file_path);
+        }
+        $this->core->getQueries()->setPollImage($poll_id, $file_path);
```

Comment:
No need to run this query if we don't have an image. Can delete the `$file_path = null`; above the if statement as well.

```suggestion
            $this->core->getQueries()->setPollImage($poll_id, $file_path);
        }
```